### PR TITLE
tracing: Fix tracing_user

### PR DESCRIPTION
**Pull request by WealianLiao**
`tracing_user.h` lost several definitions, which made CONFIG_TRACING_USER build fail.
This fixes the TRACING_USER config & adds it to the test case.

### CODE REVIEW

**Review by bbolen — APPROVED** [review]
What board were you using where you hit this?  I tested the initial commit using qemu_cortex_r5, so I am trying to figure out what I missed.

File: samples/subsys/tracing/prj_user.conf [hunk 1]
```diff
@@ -1,2 +1,3 @@
 CONFIG_TRACING=y
 CONFIG_TRACING_USER=y
+CONFIG_IDLE_STACK_SIZE=2048
```
Comment (teburd) [inline samples/subsys/tracing/prj_user.conf:3]: Why change the stack size here, this seems unrelated to the rest

**Comment by WealianLiao** [comment]
> What board were you using where you hit this? I tested the initial commit using qemu_cortex_r5, so I am trying to figure out what I missed.

Initially, I tested the feature on NPCX platform. This can be checked by this commit & revert `tracing_user.h. Then, `./scripts/twister -T ./samples/subsys/tracing/`.

File: samples/subsys/tracing/prj_user.conf (hunk 1 above)
Comment (WealianLiao) [reply samples/subsys/tracing/prj_user.conf:3]: The idle task stack is overflow on qemu_x86, qemu_cortex_m0 & qemu_riscv64. This makes the test fail. I added the description on the commit log. If this should separate to the other commit, please let me know.

**Review by teburd — APPROVED** [review]

**Review by nashif — APPROVED** [review]
